### PR TITLE
Add Basic Support for IP Tracker

### PR DESCRIPTION
**Pull request by nisdas**
**What type of PR is this?**

Feature Addition

**What does this PR do? Why is it needed?**

- [x] Tracks all observed IPs. 
- [x] Tallies count based on how many unique peer identities we see for that IP. 
- [x] Add unit test.

**Which issues(s) does this PR fix?**

N.A

**Other notes for review**

This feature for now is not included as part of our peer scorer given that it isn't fully ready. However in the future, this can be added into it and become a part of scoring.

### CODE REVIEW

File: beacon-chain/p2p/peers/status.go [hunk 1]
```diff
@@ -27,6 +27,8 @@ import (
 	"sort"
 	"time"
 
+	manet "github.com/multiformats/go-multiaddr/net"
+
```
Comment (farazdagi) [inline beacon-chain/p2p/peers/status.go:31]: Should we remove this extra blank line?

File: beacon-chain/p2p/peers/status.go [hunk 2]
```diff
@@ -579,6 +594,88 @@ func (p *Status) HighestEpoch() uint64 {
...
+	}
+	if peerData.Address == nil {
+		return false
+	}
+	ip, err := manet.ToIP(peerData.Address)
+	if err != nil {
+		return true
+	}
+	if val, ok := p.ipTracker[ip.String()]; ok {
+		if val > ColocationLimit {
```
Comment (farazdagi) [inline beacon-chain/p2p/peers/status.go:613]:
```suggestion
		if val >= ColocationLimit {
```
Non-strict inequality, probably?

Comment (nisdas) [reply beacon-chain/p2p/peers/status.go:613]: Wouldn't limit imply a strict inequality ?

File: beacon-chain/p2p/peers/status.go (hunk 1 above)
Comment (nisdas) [inline beacon-chain/p2p/peers/status.go:31]:
```suggestion
```

File: beacon-chain/p2p/peers/status.go (hunk 2 above)
Comment (farazdagi) [reply beacon-chain/p2p/peers/status.go:601]: Agree, limit is better represented by the strict inequality.

**Review by farazdagi — APPROVED** [review]

**Pull request merged**